### PR TITLE
Fix list.c iteration tools

### PR DESCRIPTION
**Pull request by phannebohm**
I found this dump in [this regression](https://libraries.openmodelica.org/branches/master-fmi/Buildings_8/files/Buildings_8_Buildings.Fluid.Movers.Validation.ControlledFlowMachine.sim). I think this should be the right implementation.

### CODE REVIEW

File: OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c [hunk 1]
```diff
@@ -83,62 +83,43 @@ void freeValueList(VALUES_LIST *valueList, unsigned int numberOfList)
...
+      /* clean list from next node */
+      infoStreamPrint(LOG_NLS_EXTRAPOLATE, 0, "cleanValueList length: %d", listLen(valueList->valueList));
+      updateNodeNext(valueList->valueList, startNode, NULL);
+      removeNodes(valueList->valueList, next);
+    }
   }
+  else listClear(valueList->valueList);
 }
 
 void cleanValueListbyTime(VALUES_LIST *valueList, double time)
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c:100]: Can you add a documentation? I've read the code and am still not completely sure what the function should do.

File: OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c [hunk 2]
```diff
@@ -83,62 +83,43 @@ void freeValueList(VALUES_LIST *valueList, unsigned int numberOfList)
 
 void cleanValueList(VALUES_LIST *valueList, LIST_NODE *startNode)
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c:84]: Can you add a documentation?

File: OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c [hunk 3]
```diff
@@ -83,62 +83,43 @@ void freeValueList(VALUES_LIST *valueList, unsigned int numberOfList)
...
-  }while(1);
-  cleanValueList(valueList, node);
-  infoStreamPrint(LOG_NLS_EXTRAPOLATE, 0, "New list length %d: ", listLen(valueList->valueList));
-  printValuesListTimes(valueList);
-  infoStreamPrint(LOG_NLS_EXTRAPOLATE, 0, "Done!");
+    listRemoveFront(valueList->valueList);
+  }
 }
 
 VALUE* createValueElement(unsigned int size, double time, double* values)
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c:125]: Documentation

**Review by AnHeuermann — CHANGES_REQUESTED** [review]
Looking good. If you have time please add some documentation to the functions you changed.

File: OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c (hunk 1 above)
Comment (phannebohm) [reply OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c:100]: I'm not entirely sure myself, I just cleaned up the usage of `LIST`s. I suppose the whole `nonlinearValuesList` needs to be refactored, but not in this PR.

**Review by AnHeuermann — APPROVED** [review]

**Pull request merged**